backend/controllers/search/prisonerSearch.js

```diff
@@ -62,10 +65,12 @@ module.exports = ({ paginationService, elite2Api, logError }) => async (req, res
       }))
 
     return res.render('prisonerSearch/prisonerSearch.njk', {
-      alertOptions: alertFlagValues.map(alertFlag => ({
-        value: alertFlag.alertCodes,
-        text: alertFlag.label,
-        checked: alertFlag.alertCodes.some(alert => alerts && alerts.includes(alert)),
+      alertOptions: alertFlagValues.map(({ alertCodes, label }) => ({
+        value: alertCodes,
+        text: label,
+        checked: multipleAlertsSelected
+          ? selectedAlerts.some(alert => alertCodes.includes(alert))
+          : Boolean(selectedAlerts) && alertCodes.includes(selectedAlerts.split(',')[0]),
```

Comment:
Nice, took me a while to understand this. The split is because the alerts in the query could be `RTP,RLG` , so even though it's one checkbox (so not an array, not multipleAlertsSelected), it could be multiple alert codes. Right?

Reply:
Yeah it's not the nicest to be fair, felt like there was a better option but it wouldn't come to me.  Going to try what @andrewrlee suggested to remove some of the complexity.